engine/src/scene.ts

```diff
@@ -92,6 +92,8 @@ export class Scene {
 
     // Remove the entity with the given name from the scene
     deleteEntity(name: string) {
+        this.entities.get(name)?._delete();
+        this.entities.delete(name);
         // Remove the entity from the systems that require it
         const entity = this.entities.get(name);
```

Comment:
Isn't this attempting to access an entity which was just deleted?

Reply:
Oh yeah, good spot